### ISSUE TITLE
Wrong Line numbers are giving in case of /

### ISSUE THREAD

**Issue by vineelavinnakota** [dotnet/roslyn#70636]
System.Diagnostics.Process.Start("https://qa-az.hxpro.hexagonseco.com/api-docs/");
/System.Windows.Forms.MessageBox.Show("help");

this is the code snippet im trying to compile using rosyln 
we are forming a syntax tree with above code along with references and namespaces.
**When it compiled it is throwing wrong line number in case of /**
 for example my complete code is 
    using System;
   System.Diagnostics.Process.Start("https://qa-az.hxpro.hexagonseco.com/api-docs/");
/System.Windows.Forms.MessageBox.Show("help");

lets assume all the references are there
my Diagnostic error should give me error at line number 3 and it is giving 
but if have any other error in the / line for example in the same line i havent referred system.windows.forms error is at line number 3 but it is giving line number 4 
for every other errors it is giving line number + 1 as the location 
for / it is giving line number as location plz tell me what to do
it is urgent 

**If commented below line has error it is giving wrong line number in case of comments**

**Comment by jaredpar** [comment]
Sorry but I'm having trouble understanding what is going wrong here. The description has a few incomplete sentences which cut off the observed behavior, it's hard to understand what the code in question is here (a few sentences say assume references, others talk about whether references are present or not). 

Can you try restating the question a bit: 
1. Give a complete code sample outlined with code blocks
2. Be clear about how the project is being built. possibly post the projcet file contents
3. Clearly state the expected and observed behaviors.

**Comment by vineelavinnakota** [comment]
using System;

using System.Runtime;

using System.Text.RegularExpressions;

using System.Linq;

using System.Linq.Expressions;

using System.IO;

using System.Net;

using System.Net.Http;

using System.Reflection;

using System.ComponentModel;

using System.Globalization;

using System.Collections.Concurrent;

using System.Collections.Generic;

using System.Xml;

using System.Xml.Linq;

using System.Data;

using Hexagon.SmartUITest.ErrorMgmt;

using Hexagon.SmartUITest.SUITUtilities;

using Utilities.Reporting;

using UINodes;

using Web.Control;

using OpenQA.Selenium;


namespace CodeSnippetCodeApp{
 public class CodeSnippet{
public int iterator=0;
public bool jh = true;
public string Password = @"Hofstaderkooth@23.";
public string Password_Old = @"Wolowitzcooper@22.";
public string username_emailid = @"ishika.ghosh@hexagon.com";

 public ReturnObject SnippetCode(ReturnObject GlobalRC)
{ 
ReturnObject rc = new ReturnObject();
try
{
System.Diagnostics.Process.Start("https://qa-az.hxpro.hexagonseco.com/api-docs/");
//System.Windows.Forms.MessageBox.Show("help");
**/System.Diagnostics.Process.Start("");**
SUITControl LOGIN = new SUITControl("LOGIN","RPlogin",@"$SUITTESTS\Delete_Operation\Delete_Operation.suit");
}
catch(BreakExecutionException ex){
return ex.rc;
}

catch(Exception ex){
rc.HandleError(EnumErrorType.Failure, " Exception occured while running code snippet: " +  ex.Message);
return rc;
}
return rc;
}
} 
 }


this is the code stream we are passing to rosyln to compile using 

here **codeToExecute** is the above code and 
 we are referring required dlls that are required for the codeToExecute

var tree = SyntaxFactory.ParseSyntaxTree(codeToExecute);
                List<MetadataReference> referencesList = new List<MetadataReference>();
                foreach (var reference in snippetInfo.References)
                {
                    #if !IsWindowsOS
                    string referencePath = reference.Replace("\\", "/");
                    referencesList.Add(MetadataReference.CreateFromFile(referencePath));
                    #else
                    referencesList.Add(MetadataReference.CreateFromFile(reference));
                    #endif
                }
                    
                var compilation = CSharpCompilation.Create("Executor.cs")
                    .WithOptions(new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary,
                        optimizationLevel: OptimizationLevel.Release))
                    .AddReferences(referencesList)
                    .AddSyntaxTrees(tree);

                EmitResult compilationResult = null;
                Assembly compiledAssembly = null;

                Stream codeStream = null;
                using (codeStream = new MemoryStream())
                {
                    // Actually compile the code
                    **compilationResult** = compilation.Emit(codeStream);

                    if(compilationResult.Success)
                        compiledAssembly = Assembly.Load(((MemoryStream)codeStream).ToArray());
                }



In **compilationResult**  as compiling in above code , the error info in compilation result is discussed below


**Expected output  **
 error at line number 61 bolded in the above given code, with some error code like CS1525 lets say
**Observed output**
error at (59,83),with error code
here line number is 59 and 83 is column 




not only for the / in line number 61 in codesnippet, for any other symbol it is throwing wrong line number

**Comment by vineelavinnakota** [comment]
Any update on this. plz do respond

**Comment by CyrusNajmabadi** [comment]
@vineelavinnakota your post is very confusing. Perhaps you could upload the repro to a git repo and link to that.

**Comment by vineelavinnakota** [comment]
omg, i have clearly mentioned. okay let me try it once again. ill make it simple i dont go indetail.

The below is the code snippet i want to compile using Rosyln

System.Diagnostics.Process.Start("https://qa-az.hxpro.seco.com/api-docs/");
//System.Windows.Forms.MessageBox.Show("help");
/System.Diagnostics.Process.Start("");


when i try it compile it is giving wrong line number 

the above mentioned post has my full codeim passing to rosyln with namespaces and class

**expected**
error at / with line number 3
observed
error at / with line number 1


if still confusing let me know

**Comment by vineelavinnakota** [comment]
[CodeAnalysisUsingRosyln.zip](https://github.com/dotnet/roslyn/files/13276927/CodeAnalysisUsingRosyln.zip)
please check the above attached zip file .
I have created a console app with the required code just run the console app and observe the compilation result
Thank you

**Comment by vineelavinnakota** [comment]
[CodeAnalysisUsingRosyln.zip](https://github.com/dotnet/roslyn/files/13277060/CodeAnalysisUsingRosyln.zip)
This is the sample app i have created just run the app and check the compilation result

**Comment by jaredpar** [comment]
Looking at the repro you provided I see the following code

```csharp
using System;
namespace CodeSnippetCodeApp 
{
  public class CodeSnippet
  {
    public void SnippetCode()
    { 
      try
      {
        System.Diagnostics.Process.Start("https://qa-az.hxpro.hexagonseco.com/api-docs/");
        //System.Windows.Forms.MessageBox.Show("help");
        /System.Diagnostics.Process.Start("");
      }
      catch(Exception ex)
      {
        Console.WriteLine(ex.Message.ToString());
      }
    }
  } 
}
```

The compiler emits the following diagnostic for this code 

> (8,83): error CS1525: Invalid expression term '/'

That is the diagnostic I would expect here given that `/` cannot lead an expression.

**Comment by vineelavinnakota** [comment]
okay, thank you
